class/class5.1/MegaDataPool.cpp

```diff
@@ -0,0 +1,41 @@
+#include "MegaDataPool.hpp"
+
+MegaDataPool::MegaDataPool(size_t poolSize) : poolSize(poolSize), usedSize(0) {
+    for (size_t i = 0; i < poolSize; ++i) {
+        pool.push(new MegaData());
+    }
+}
+
+MegaDataPool::~MegaDataPool() {
+    while (!pool.empty()) {
+        delete pool.front();
+        pool.pop();
+    }
+}
+
+MegaData* MegaDataPool::acquire() {
+    if (pool.empty()) {
+        throw std::runtime_error("No available objects in the pool");
+    }
+    MegaData* obj = pool.front();
+    pool.pop();
```

Comment:
let's avoid pop of data. we can just track them via some container with flags used/free.